### DISCUSSION TITLE
Is there a way to organize the specific wallets we display?

### DISCUSSION THREAD

**Discussion by castig** [WalletConnect/walletconnect-monorepo#3416] [Developer Support] [answered]
Hi! We're using Wallet Connect on [Console.xyz](https://www.console.xyz/). We'd like to prioritize the most popular wallets at the top — but right now (for example) MetaMask is down in the third row of the list. 

Is it possible that we could arrange these ourselves? Or provide alternative UIs to highlight certain more popular wallets? 

<img width="365" alt="Screenshot 2023-08-09 at 5 01 41 PM" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/237017/d9a99d8a-4429-48aa-8ddd-eb89e20476a3">
<img width="379" alt="Screenshot 2023-08-09 at 5 01 34 PM" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/237017/c8101b56-33d7-47d2-8af5-b8bdb5feeea8">

**Comment by glitch-txs** [comment]
Hi you can use this in the `qrModalOptions` when initiating the provider
https://docs.walletconnect.com/2.0/web3modal/react/wagmi/options#walletimages-optional

**Reply by sachin-kiwi** [reply]
Link says page not found

**Reply by MuckT** [reply, marked as answer]
This link should work: https://docs.walletconnect.com/2.0/web3modal/react/wagmi/options#walletimages-optional.

**Reply by MuckT** [reply]
You can use the following code to add whatever wallets you want first:
```JavaScript
<Web3Modal
  projectId={projectId} 
  ethereumClient={ethereumClient}
  explorerRecommendedWalletIds={[
    // Valora Wallet ID https://walletconnect.com/explorer/valora
    'd01c7758d741b363e637a817a09bcf579feae4db9f5bb16f599fdd1f66e2f974',
    // Coinbase Wallet ID https://walletconnect.com/explorer/coinbase-wallet
    'fd20dc426fb37566d803205b19bbc1d4096b248ac04548e3cfb6b3a38bd033aa',
    // Trust Wallet ID https://walletconnect.com/explorer/trust-wallet
    '4622a2b2d6af1c9844944291e5e7351a6aa24cd7b23099efac1b2fd875da31a0',
  ]}
/>
```

**Comment by castig** [comment]
Thanks! Do you know what determines the default list order? It seems odd.
Is it supposed to be most popular, or is it random?

On Wed, Aug 9, 2023 at 9:08 PM Glitch ***@***.***> wrote:

> Hi you can use this in the qrModalOptions when initiating the provider
>
> https://docs.walletconnect.com/2.0/web/walletConnectModal/modal/options#explorerrecommendedwalletids-optional
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/WalletConnect/discussions/3416#discussioncomment-6685836>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABZ3WPFFV7RDHAWNFTF3WLXUQYCDANCNFSM6AAAAAA3KRZIUA>
> .
> You are receiving this because you authored the thread.Message ID:
> <WalletConnect/walletconnect-monorepo/repo-discussions/3416/comments/6685836
> @github.com>
>


-- 
Chris Castiglione | castig.org  |  @castig <http://www.twitter.com/castig>
Teacher at OneMonth.com <https://onemonth.com/>

**Reply by Yangjiajiaa** [reply]
I have the same question as well.
